### PR TITLE
PEP-8 style fixes.

### PR DESCRIPTION
**Pull request by jmchilton**
- Setup a tox.ini for linting code using Python 2.7.

TODO: Still 5 linting problems, 4 are actual programming errors.

cwltool/draft2tool.py:159:53: F812 list comprehension redefines 'a' from line 75
cwltool/main.py:356:33: F821 undefined name 'processobj'
cwltool/main.py:356:45: F821 undefined name 'ctx'
cwltool/main.py:360:33: F821 undefined name 'processobj'
cwltool/main.py:360:45: F821 undefined name 'ctx'

### CODE REVIEW

**Comment by tetron** [comment]
Do you want to fix the other linting errors or shall I do that?

**Comment by jmchilton** [comment]
I can take a stab at it, but if the fixes had been obvious I would have made them already. I think it can be merged as is and we can work on these problems later - there are some import fixes and stuff in here that actually were bugs and not just style "issues".

File: cwltool/cwltest.py [hunk 1]
```diff
@@ -92,7 +94,7 @@ def run_test(args, i, t):
         _logger.error(outstr)
         _logger.error("Parse error %s", str(e))
 
-    pwd = os.path.abspath(os.path.dirname(t["job"]))
+    os.path.abspath(os.path.dirname(t["job"]))
```
Comment (tetron) [inline cwltool/cwltest.py]: That's interesting, I think this line and the comment block below it can go away entirely

File: cwltool/workflow.py [hunk 2]
```diff
@@ -302,8 +300,8 @@ def __init__(self, toolpath_object, **kwargs):
         kwargs["requirements"] = self.requirements
         kwargs["hints"] = self.hints
 
-        makeTool = kwargs.get("makeTool")
-        self.steps = [WorkflowStep(step, n, **kwargs) for n,step in enumerate(self.tool.get("steps", []))]
+        kwargs.get("makeTool")
```
Comment (tetron) [inline cwltool/workflow.py]: Another spurious line that can probably go away

**Comment by tetron** [comment]
*sigh* I see many merge conflicts in my future.

By "actual bugs" do you mean things like spurious imports?

Did you try running the conformance tests?   E.g. "common-workflow-language/run_tests.sh RUNNER=cwltool"

**Comment by tetron** [comment]
Hi @jmchilton any chance you can re-run the linting and update the pull request?

**Comment by tetron** [comment]
@jmchilton or at least tell me what you did so I can run it myself.

**Comment by jmchilton** [comment]
Hey @tetron, been busy - I'll return to this and rebase. Now that I realize I can test it with the conformance tests, I'll let you know when it is working and ready to go.

**Comment by jmchilton** [comment]
I'll retry a much smaller first step toward something like this. Sorry for opening something so unwieldy and untested and keeping it open.

**Pull request closed without merging**